packages/grid/x-data-grid/src/hooks/features/selection/useGridSelection.ts

```diff
@@ -265,6 +266,12 @@ export const useGridSelection = (
    * EVENTS
    */
   const removeOutdatedSelection = React.useCallback(() => {
+    if (props.keepNonExistentRowsSelected) {
+      // Trigger the selection model callback again but with whatever values are currently selected
+      // This will help consumers keep the selectionModel in sync
+      apiRef.current.setSelectionModel(Object.values({ ...selectedIdsLookupSelector(apiRef) }));
```

Comment:
Why you added this call? Was it added to help testing? Since `Object.values` always returns a new object, `setSelectionModel` will be called even if the model didn't change. Eventually, you'll need to pass `selectionModel` to clean the selection, so this call might not be necessary.